### PR TITLE
Use '\n' instead of std::endl.

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This removes most instances of `std::endl` from Mantid (See our discussion in the tips channel on Slack and [this video](http://articles.emptycrate.com/2016/04/18/stop_using_std_endl.html)). I tried appending `\n` to the end of a string literal and using  `'\n'` otherwise. 

**To test:**

I made most of these changes with find and replace, so they should be quite consistent and many potential issues will cause the build to fail. I would suggest reviewing until you are comfortable with the substitutions. The last few commits were done by hand and therefore more likely to contain errors. You may want to review them more carefully.

This PR has no associated GitHub issue

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by quantumsteve** [comment]
@martyngigg

File: Framework/CurveFitting/src/Algorithms/RefinePowderInstrumentParameters.cpp [hunk 1]
```diff
@@ -295,8 +295,7 @@ void RefinePowderInstrumentParameters::fitInstrumentParameters() {
   for (size_t i = 0; i < m_dataWS->readX(0).size(); ++i)
     outss << m_dataWS->readX(0)[i] << "\t\t" << m_dataWS->readY(0)[i] << "\t\t"
           << m_dataWS->readE(0)[i] << '\n';
-  cout << "Input Peak Position Workspace To Fit: " << endl
-       << outss.str() << '\n';
+  cout << "Input Peak Position Workspace To Fit: \n" << outss.str() << '\n';
```
Comment (martyngigg) [inline Framework/CurveFitting/src/Algorithms/RefinePowderInstrumentParameters.cpp]: Should we get rid of any usages of  `cout`/`cerr` like this? They don't seem like a good idea in the middle of an algorithm.

**Comment by martyngigg** [comment]
The changes look good. 

I notice now that we have a lot of places where we are using the the stream operators to log a single string literal. This would be far more efficient if it were passed directly to the appropriate `Logger` method, e.g `g_log.notice("My string literal")` rather than `g_log.notice() << "My string literal\n"`. Do you think this is worth addressing?

**Comment by quantumsteve** [comment]
@martyngigg Yes, changes like that would be beneficial, but think it is outside the scope of this pull request. I added your comments and a few other ideas about logging to the maintenance tasks for another release.

**Comment by martyngigg** [comment]
Agreed.

:shipit:

**Pull request merged**